File: solutions/r/nth-prime/2/nth-prime.R
prime <- function(n) {
  if(n<1)  stop("There is no zeroth prime")
  if(n==1) return(2)
  cand<-3
  i<-2
  while(i<n){
    cand<-cand+2
    if(all(cand %% 3 : ceiling(sqrt(cand)) !=0)) i<-i+1
  }
  cand
}

# inspired by Steffan153's solution

File: solutions/dart/nth-prime/2/lib/nth_prime.dart
class NthPrime {
  
  int prime(int N){
    if(N<1)
      throw ArgumentError("There is no zeroth prime");
    if(N<3)
      return N==1 ? 2 : 3; 
    
    var primes = [2,3];
    for(var cand=5; primes.length<=N; cand += 2){
      var isPrime = true;
      for(var p in primes){
        if(0==cand % p){
          isPrime = false;
          break;
        }        
      }
      if(isPrime)
        primes.add(cand);
    }
    return primes[N-1];
  }
  
}
